internal/gps/source_manager.go

```diff
@@ -281,31 +280,15 @@ func (sm *SourceMgr) HandleSignals(sigch chan os.Signal) {
 				// Set up a timer to uninstall the signal handler after three
 				// seconds, so that the user can easily force termination with a
 				// second ctrl-c
-				go func(c <-chan time.Time) {
-					<-c
+				time.AfterFunc(3*time.Second, func() {
 					signal.Stop(sch)
-				}(time.After(3 * time.Second))
+				})
 
-				if !atomic.CompareAndSwapInt32(&sm.releasing, 0, 1) {
-					// Something's already called Release() on this sm, so we
-					// don't have to do anything, as we'd just be redoing
-					// that work. Instead, deregister and return.
-					return
-				}
```

Comment:
This block was setting the atomic prior to the `count()` call, which ensured that no more ops launch, fixing the count. 
 Currently, more ops could launch between the `count()` call and the atomic swap at the start of `Release()`.  What about moving this count/log line inside of `Release()`, just after the atomic swap?

Reply:
I suppose the 'signal received' portion of the message won't apply from inside the method, so maybe that should still be logged separately.

Reply:
> This block was setting the atomic prior to the `count()` call, which ensured that no more ops launch, fixing the count. Currently, more ops could launch between the `count()` call and the atomic swap at the start of `Release()`. What about moving this count/log line inside of `Release()`, just after the atomic swap?

(just noting that this sounds like the class of concern that motivated the original design)

Reply:
@tamird thoughts on this?

Reply:
We could put that print statement inside the Release method - would that suit you?

Reply:
actually, let's keep the print statement outside, where it is.

in general, we try to treat gps like a pure library within dep. for the most part, gps isn't aware dep exists - it only shares some low-level filesystem manipulation methods, and some test helpers. also to that end, we try really, really hard to avoid direct print statements in gps itself. this one print statement is the sole place, in all of gps, where we actually print directly, and i only let it in there because it would have required too much refactoring to arrange the signal handling otherwise.

so yeah, let's call this PR a victory on its own terms, and we can think about ways of refactoring for a more controlled, informative shutdown experience later.